funnel/views/participant.py

```diff
@@ -48,64 +51,83 @@ def participant_badge_data(participants, project):
+        'puk': participant.puk,
+        'fullname': participant.fullname,
+        'job_title': participant.job_title,
+        'company': participant.company,
+        'project_id': project_id,
+    }
     if full:
-        return {
-            '_id': participant.id,
-            'puk': participant.puk,
-            'fullname': participant.fullname,
-            'job_title': participant.job_title,
-            'company': participant.company,
-            'email': participant.email,
-            'twitter': participant.twitter,
-            'phone': participant.phone,
-            'project_id': project_id,
-            'space_id': project_id,  # FIXME: Remove when the native app switches over
-        }
-    else:
-        return {
-            '_id': participant.id,
-            'puk': participant.puk,
-            'fullname': participant.fullname,
-            'job_title': participant.job_title,
-            'company': participant.company,
-            'project_id': project_id,
-            'space_id': project_id,  # FIXME: Remove when the native app switches over
-        }
+        data.update(
+            {
+                'email': participant.email,
+                'twitter': participant.twitter,
+                'phone': participant.phone,
+            }
+        )
+    return data
 
 
 def participant_checkin_data(participant, project, event):
-    return {
-        'pid': participant.id,
+    psuuid = uuid2suuid(participant.uuid)
+    data = {
+        'psuuid': psuuid,
         'fullname': participant.fullname,
         'company': participant.company,
         'email': mask_email(participant.email),
         'badge_printed': participant.badge_printed,
         'checked_in': participant.checked_in,
         'ticket_type_titles': participant.ticket_type_titles,
     }
+    if 'checkin_event' in current_auth.user.current_permissions:
+        data.update(
+            {
+                'badge_url': url_for(
+                    'ParticipantView_badge',
+                    profile=project.profile.name,
+                    project=project.name,
+                    suuid=psuuid,
+                ),
+                'label_badge_url': url_for(
+                    'ParticipantView_label_badge',
+                    profile=project.profile.name,
+                    project=project.name,
+                    suuid=psuuid,
+                ),
+                'edit_url': url_for(
+                    'ParticipantView_edit',
+                    profile=project.profile.name,
+                    project=project.name,
+                    suuid=psuuid,
+                ),
+            }
+        )
+    return data
 
 
 @route('/<profile>/<project>/participants')
 class ProjectParticipantView(ProjectViewMixin, UrlForView, ModelView):
     __decorators__ = [legacy_redirect]
 
     @route('json')
+    @render_with(json=True)
     @lastuser.requires_login
     @requires_permission('view')
     def participants_json(self):
```

Comment:
This view should be dropped. It is a security risk. It was added for speedy badge scanning in the native app.